pkg/cmd/adm/install_operator.go

```diff
@@ -0,0 +1,200 @@
+	catalogSourceKey := types.NamespacedName{Name: fmt.Sprintf("source-%s-operator", operator), Namespace: args.namespace}
+	catalogSource := newCatalogSource(catalogSourceKey, operator)
+	kubeClient, err := newKubeClient(ctx, args.kubeConfig)
+	if err != nil {
+		return err
+	}
+	if err := kubeClient.Create(ctx, catalogSource); err != nil {
+		return err
+	}
+	if err := waitUntilCatalogSourceIsReady(ctx.CommandContext, kubeClient, catalogSourceKey, timeout); err != nil {
+		return err
+	}
+	ctx.Printlnf("CatalogSource %s is ready", catalogSourceKey)
+
+	// install operator group
+	operatorGroup := newOperatorGroup(types.NamespacedName{Name: fmt.Sprintf("og-%s-operator", operator), Namespace: args.namespace})
+	if err := kubeClient.Create(ctx, operatorGroup); err != nil {
+		return err
+	}
+
+	// install subscription
+	subscription := newSubscription(types.NamespacedName{Name: fmt.Sprintf("subscription-%s-operator", operator), Namespace: args.namespace}, fmt.Sprintf("toolchain-%s-operator", operator), catalogSourceKey.Name)
+	if err := kubeClient.Create(ctx, subscription); err != nil {
+		return err
+	}
+	if err := waitUntilInstallPlanIsComplete(ctx.CommandContext, kubeClient, args.namespace, timeout); err != nil {
+		return err
+	}
+	ctx.Println(fmt.Sprintf("InstallPlans for %s-operator are ready", operator))
+	return nil
+}
+
+func newCatalogSource(name types.NamespacedName, operator string) *olmv1alpha1.CatalogSource {
+	return &olmv1alpha1.CatalogSource{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: name.Namespace,
+			Name:      name.Name,
+		},
+		Spec: olmv1alpha1.CatalogSourceSpec{
+			SourceType:  olmv1alpha1.SourceTypeGrpc,
+			Image:       fmt.Sprintf("quay.io/codeready-toolchain/%s-operator-index:latest", operator),
+			DisplayName: "Dev Sandbox Operators",
+			Publisher:   "Red Hat",
+			UpdateStrategy: &olmv1alpha1.UpdateStrategy{
+				RegistryPoll: &olmv1alpha1.RegistryPoll{
+					Interval: &metav1.Duration{
+						Duration: 1 * time.Minute,
+					},
+				},
+			},
+		},
+	}
+}
+
+func newOperatorGroup(name types.NamespacedName) *olmv1.OperatorGroup {
+	return &olmv1.OperatorGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: name.Namespace,
+			Name:      name.Name,
+		},
+		Spec: olmv1.OperatorGroupSpec{
+			TargetNamespaces: []string{
+				name.Namespace,
+			},
+		},
+	}
+}
+
+func newSubscription(name types.NamespacedName, operatorName, catalogSourceName string) *olmv1alpha1.Subscription {
+	return &olmv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: name.Namespace,
+			Name:      name.Name,
+		},
+		Spec: &olmv1alpha1.SubscriptionSpec{
+			Channel:                "staging",
+			InstallPlanApproval:    olmv1alpha1.ApprovalAutomatic,
+			Package:                operatorName,
+			CatalogSource:          catalogSourceName,
+			CatalogSourceNamespace: name.Namespace,
+		},
+	}
+}
+
+func newKubeClient(ctx *extendedCommandContext, kubeConfigPath string) (cl runtimeclient.Client, err error) {
```

Comment:
`register-member` command contains a nearly identical function. I assume we're going to see a lot more of this once we drop support for initializing the connection using just a token and start relying on the kubeconfig in all the commands. Maybe it is a good time to introduce some abstraction now?

Reply:
Yes, this is also something we have in host-operator/toolchain-e2e I believe. I wanted to do it later, but I can do it right now 👍

Reply:
I have extracted and refactored a bit the function, please see https://github.com/kubesaw/ksctl/pull/48/commits/887c79495617af1f405addbdee911d1cdd837fc2

We can refactor the other commands ( `register-member` ) and eventually move it toolchain-common later so it can be reused outside of this repo as well.